test/crd_management_test.go

```diff
@@ -118,63 +105,45 @@ var _ = Describe("CRD management tests", func() {
 
 		mgr = nil
 
-		_ = c.Delete(context.Background(), npCRD.DeepCopy())
+		//Need to make sure the operator is shut down prior to deleting and recreating the networkpolicies CRD otherwise
+		//the controller initialized within the tests can recreate the CRD between the deletion confirmation and recreation
+		//attempt, creating a timing window where failures can occur.
+		cancel()
 		Eventually(func() error {
-			u := &unstructured.Unstructured{}
-			u.SetGroupVersionKind(schema.GroupVersionKind{
-				Group:   "apiextensions.k8s.io",
-				Version: "v1",
-				Kind:    "CustomResourceDefinition",
-			})
-
-			k := client.ObjectKey{Name: npCRD.Name}
-			err := c.Get(context.Background(), k, u)
-			return err
-		}, 20*time.Second).ShouldNot(BeNil())
-		npCRD.SetResourceVersion("")
-		err := c.Create(context.Background(), npCRD)
+			select {
+			case <-operatorDone:
+				return nil
+			default:
+				return fmt.Errorf("operator did not shutdown")
+			}
+		}, 60*time.Second).Should(BeNil())
+
+		err := c.Delete(context.Background(), npCRD)
 		Expect(err).NotTo(HaveOccurred())
+		// The "MustPassRepeatedly" here is important because there is some jitter that isn't fully understood that can
+		// cause flakes without it
 		Eventually(func() error {
-			u := &unstructured.Unstructured{}
-			u.SetGroupVersionKind(schema.GroupVersionKind{
-				Group:   "apiextensions.k8s.io",
-				Version: "v1",
-				Kind:    "CustomResourceDefinition",
-			})
-
-			k := client.ObjectKey{Name: npCRD.Name}
-			err := c.Get(context.Background(), k, u)
-			return err
-		}, 20*time.Second).Should(BeNil())
+			err := GetResource(c, npCRD)
+			if kerror.IsNotFound(err) || kerror.IsGone(err) {
+				return nil
+			} else if err != nil {
+				return err
+			} else {
+				return fmt.Errorf("NetworkPolicy CRD still exists")
+			}
+		}, 20*time.Second).MustPassRepeatedly(50).ShouldNot(HaveOccurred())
```

Comment:
Hm, do we really need to run this 50 times?

Reply:
50 might be a touch aggressive. When I was testing I observed that sometimes the first ~10 iterations could toggle back and forth between success and failure so I just wanted to ensure that the value had truly stabilized prior to advancing. I could try reducing this and running longevity tests but ultimately this is only requiring a 500 ms stable result in exchange for completely eliminating any flakiness.